src/db.c

```diff
@@ -351,7 +266,8 @@ int dbAddRDBLoad(redisDb *db, sds key, robj *val) {
  *
  * The program is aborted if the key was not already present. */
 static void dbSetValue(redisDb *db, robj *key, robj *val, int overwrite, dictEntry *de) {
-    if (!de) de = dbFind(db, key->ptr, DB_MAIN);
+    int slot = getKeySlot(key->ptr);
+    if (!de) de = kvstoreDictFind(db->keys, slot, key->ptr);
```

Comment:
i wanted to avoid getKeySlot

the caching only works if `server.current_client->slot` is valid, if it's not, two calls to getKeySlot will actually calculate the hash twice